Add rt1180 basic environment

lucien-nxp opened the pull request:
Enable RT1180 basic environment:
Test below cases on cm33/cm7 cores:
1. hello_world
2. button
3. blinky
4. synchronization
5. philosophers

decsny reviewed (dismissed):
Please see https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#multiple-commits-on-a-single-pr

Need to clean up your commit history on this PR

danieldegrasse commented on boards/nxp/mimxrt1180_evk/mimxrt1180_evk_mimxrt1189_cm33_defconfig:
```diff
@@ -0,0 +1,14 @@
+CONFIG_ARM_MPU=y
+CONFIG_HW_STACK_PROTECTION=y
+CONFIG_PINCTRL=y
+CONFIG_FLEXSPI_CONFIG_BLOCK_OFFSET=0x400
```
Do we need to set this again? It is set at SOC level

lucien-nxp replied:
No need, I will delete it. Sorry, this content was copied from RT1170.

danieldegrasse commented on boards/nxp/mimxrt1180_evk/doc/index.rst:
```diff
@@ -0,0 +1,235 @@
+The MIMXRT1180 SoC has 12 UARTs. One is configured for the console and the
+remaining are not used.
+
+Programming and Debugging
```
We should add some blurb here about how to to load an image onto the CM7 core versus the CM33 core

lucien-nxp replied:
I have added detailed operation in Using J-Link and Using linkserver section, because there are some difference on two ways.

danieldegrasse commented:
As far as the compliance failures, an easy way to check for any in your PR before pushing is to run the following (on linux) from your Zephyr directory, with the PR branch checked out:

```
git diff $(git merge-base main HEAD) | ./scripts/checkpatch.pl
```

lucien-nxp commented on soc/nxp/imxrt/Kconfig.defconfig:
```diff
@@ -75,6 +76,7 @@ config SYSTEM_WORKQUEUE_STACK_SIZE
 endif # ETH_NXP_ENET
 
 config MEMC
+	default n if SOC_SERIES_IMXRT118X
```
Yes, there is no flexram peripheral on RT1180 soc. I will fix it like above.

lucien-nxp replied:
Fixed.

lucien-nxp commented on soc/nxp/imxrt/Kconfig.defconfig:
```diff
@@ -7,14 +7,15 @@ if SOC_FAMILY_NXP_IMXRT
 
 config ROM_START_OFFSET
 	default 0x400 if BOOTLOADER_MCUBOOT
+	default 0xb000 if BOOT_FLEXSPI_NOR && SOC_SERIES_IMXRT118X
```
For Q1: I can comment on this Kconfig to allow users to configure the image start address they want to.
For Q2: 0xA000 already provides enough space for users to install additional container headers.  Actually, ROM only support at most two container headers. Even if the second container header is used, the first container must be for NXP firmware.

lucien-nxp commented on boards/nxp/mimxrt1180_evk/doc/mimxrt1180_evk.jpg:
. webp have been updated on new commit.

lucien-nxp commented on boards/nxp/mimxrt1180_evk/board.cmake:
```diff
@@ -0,0 +1,21 @@
+board_runner_args(jlink "--device=MIMXRT1189xxx8_M7")
+endif()
+
+include(${ZEPHYR_BASE}/boards/common/jlink.board.cmake)
```
No, this content was copied from RT1170. Sorry, I don't know first one represents the default debug mode, which I will modify later.

danieldegrasse replied:
No worries- it isn't documented well (or anywhere?), so very easy to miss

lucien-nxp commented on soc/nxp/imxrt/imxrt118x/Kconfig:
```diff
@@ -0,0 +1,56 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_IMXRT118X
```
Yes, I agree.

lucien-nxp commented on soc/nxp/imxrt/imxrt118x/Kconfig.defconfig:
```diff
@@ -0,0 +1,34 @@
+
+endif # CORTEX_M_SYSTICK
+
+config DCDC_VALUE
```
Copied from RT1170, I will delete it.

lucien-nxp commented on dts/arm/nxp/nxp_rt118x_cm7.dtsi:
```diff
@@ -0,0 +1,113 @@
+		};
+		flexspi1: spi@425e0000 {
+			reg = <0x425e0000 0x4000>,<0x28000000 DT_SIZE_M(16)>;
+		};
```
Flexspi2 will be added in the future PR I think, don't be used currently.

lucien-nxp commented on boards/nxp/mimxrt1180_evk/board.cmake:
```diff
@@ -0,0 +1,21 @@
+# SPT can be downloaded on NXP web: https://www.nxp.com/design/design-center/software/development-software/mcuxpresso-software-and-tools-/mcuxpresso-secure-provisioning-tool:MCUXPRESSO-SECURE-PROVISIONING
+# Details about the usage of SPT on MIMXRT1180-EVK board can be referred on chapter 7 of getting start with Mcuxpresso SDK for MIMXRT1180-EVK doc in SDK package.
+if(CONFIG_SOC_MIMXRT1189_CM33)
+#Note: Please ensure the SW5 on "0100" mode
```
These settings have been recorded in index.rst. I can remove it.

lucien-nxp commented on dts/arm/nxp/nxp_rt118x_cm7.dtsi:
```diff
@@ -0,0 +1,113 @@
+			ranges = <0x0 0x40000000 0x10000000>;
+		};
+		flexspi1: spi@425e0000 {
+			reg = <0x425e0000 0x4000>,<0x28000000 DT_SIZE_M(16)>;
```
Sorry, I thought this was describing the space size of the external flash. Will fix it.

lucien-nxp commented on soc/nxp/imxrt/imxrt118x/soc.c:
```diff
@@ -0,0 +1,252 @@
+{
+	SystemInit();
+
+#if defined(FLEXRAM_RUNTIME_BANKS_USED)
```
Yes, I will remove it.

lucien-nxp committed:
dts: arm: nxp: add dts files for RT1180

The offset of the peripheral is abstracted so that
the peripheral can be defined in RT118x.dtsi,that
is a common dtsi file for RT1180.

Due to cm33 core, add ns/s files which are served
on different status for cm33 core.

Add rt118x_cm7 dtsi file

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

lucien-nxp committed:
drivers: clock_control: update clock_control/clock_control_mcux_ccm_rev2.c

Due to two lpuart instance use common clock root,
update driver to support new feature for RT1180.

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

lucien-nxp committed:
drivers: gpio: update rgpio driver code

RGPIO model on RT1180 is same with igpio, however,
current driver can't support.

Update gpio driver to support gpio pad control

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

lucien-nxp committed:
drivers: pinctrl: update pinctrl/pinctrl_imx.c

Add code enable rt1180 iomuxc clock

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

lucien-nxp committed:
boards: nxp: add mimxrt1180_evk board

Support rgpio/lpuart peripherals
Support linkserver/jlink debug ways
Test cases: hello_world; button; blinky; synchronization; philosophers

There is one limitation with flashing and debugging the SoC's second
core(M7) that will be addressed with furture submission

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

lucien-nxp committed:
soc: nxp: imxrt: add imxrt118x series and update related files

Add new RT118x device, due to structure is different from rt11xx series.
Boot header also differ from rt10xx and rt11xx, so add support for boot
container.

define new container name and new container tag

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

lucien-nxp commented on soc/nxp/imxrt/container.ld:
```diff
@@ -0,0 +1,11 @@
+#if defined(CONFIG_SOC_SERIES_IMXRT118X)
+__CONTAINER_IMG_OFFSET = 0xA000;
+# define the image size directly. If image is larger than 0x100000, macro need to be modified
+__CONTAINER_IMG_SIZE = 0x100000;
```
Fixed

lucien-nxp commented on soc/nxp/imxrt/imxrt118x/Kconfig:
```diff
@@ -0,0 +1,56 @@
+	select ARM
+	select CLOCK_CONTROL
+	select HAS_MCUX_CACHE
+    select ARMV8_M_DSP if CPU_CORTEX_M33
```
Fixed

danieldegrasse replied:
I wasn't clear enough here, I apologize! What I meant was something like this:

```
config SOC_MIMXRT1189_CM33
	select CPU_CORTEX_M33
        select ARMV8_M_DSP

config SOC_MIMXRT1189_CM7
	select CPU_CORTEX_M7
        select CPU_HAS_FPU_DOUBLE_PRECISION 
```

Then drop those `select x if SOC_xxx` statements from the `config SOC_SERIES_IMXRT118X` segment

lucien-nxp commented on soc/nxp/imxrt/imxrt118x/Kconfig:
```diff
@@ -0,0 +1,56 @@
+	select CPU_HAS_ICACHE if CPU_CORTEX_M7
+	select CPU_HAS_DCACHE if CPU_CORTEX_M7
+	select CPU_HAS_FPU
+	select CPU_HAS_FPU_DOUBLE_PRECISION if CPU_CORTEX_M7
```
Fixed

lucien-nxp commented on soc/nxp/imxrt/imxrt118x/Kconfig:
```diff
@@ -0,0 +1,56 @@
+	select CPU_HAS_ARM_MPU
+	select INIT_ARM_PLL
+	select CPU_HAS_ICACHE if CPU_CORTEX_M7
+	select CPU_HAS_DCACHE if CPU_CORTEX_M7
```
Fixed.

danieldegrasse commented on boards/nxp/mimxrt1180_evk/doc/index.rst:
```diff
@@ -0,0 +1,251 @@
+
+
+Using linkserver
+---------------------------------
```
Underlines need to be aligned with the header above
```suggestion
Using linkserver
----------------
```

danieldegrasse commented on boards/nxp/mimxrt1180_evk/doc/index.rst:
```diff
@@ -0,0 +1,251 @@
+EVK.
+
+
+Using linkserver
```
```suggestion
Using Linkserver
```

danieldegrasse commented on scripts/west_commands/runners/jlink.py:
I think we probably need to split this (useful) addition into a separate PR- since it affects common code and is only related to this PR in that it adds functionality the PR then depends on

lucien-nxp replied:
New PR created：https://github.com/zephyrproject-rtos/zephyr/pull/74226

danieldegrasse commented on west.yml:
Commit updating HAL should be first in this PR commit series, since other commits depend on it

lucien-nxp replied:
Place in the first commit

decsny replied:
please put it in the soc commit, not a separate commit

danieldegrasse commented on include/zephyr/linker/section_tags.h:
The commit adding section tags needs to be prior to the commit adding the RT1180 SOC support

lucien-nxp replied:
Done

decsny replied:
please add it to the soc commit rather than a separate commit

hakehuang commented:
build failure


<html>
<body>

CASE_NAME | ZEPHYR_VERSION | RESULT | PLATFORM | TEST_PATH
-- | -- | -- | -- | --
sample.drivers.flash.shell | v3.6.0-4994-g6deb75bae2a5 | Build failure | mimxrt1180_evk/mimxrt1189/cm33 | samples/drivers/flash_shell
mgmt.mcumgr.fs.mgmt.hash.supported.crc32 | v3.6.0-4994-g6deb75bae2a5 | Build failure | mimxrt1180_evk/mimxrt1189/cm33 | tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported
mgmt.mcumgr.fs.mgmt.hash.supported.sha256 | v3.6.0-4994-g6deb75bae2a5 | Build failure | mimxrt1180_evk/mimxrt1189/cm33 | tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported
mgmt.mcumgr.fs.mgmt.hash.supported.all | v3.6.0-4994-g6deb75bae2a5 | Build failure | mimxrt1180_evk/mimxrt1189/cm33 | tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported


</body>
</html>

hakehuang commented:
compilation error

<html>
<body>

CASE_NAME | ZEPHYR_VERSION | RESULT | PLATFORM | TEST_PATH
-- | -- | -- | -- | --
sample.drivers.flash.shell | v3.6.0-4994-g6deb75bae2a5 | Build failure | mimxrt1180_evk/mimxrt1189/cm7 | samples/drivers/flash_shell
mgmt.mcumgr.fs.mgmt.hash.supported.crc32 | v3.6.0-4994-g6deb75bae2a5 | Build failure | mimxrt1180_evk/mimxrt1189/cm7 | tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported
mgmt.mcumgr.fs.mgmt.hash.supported.sha256 | v3.6.0-4994-g6deb75bae2a5 | Build failure | mimxrt1180_evk/mimxrt1189/cm7 | tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported
mgmt.mcumgr.fs.mgmt.hash.supported.all | v3.6.0-4994-g6deb75bae2a5 | Build failure | mimxrt1180_evk/mimxrt1189/cm7 | tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported

</body>
</html>

hakehuang commented:
besides, build failure. there are many init failures.

and below application will cause the rt1180 locked

tests/posix/env/portability.posix.env.newlib

```

2024-06-13 11:58:41,837 - twister - ERROR - mimxrt1180_evk/mimxrt1189/cm33 tests/posix/env/portability.posix.env.newlib       ^[[31m FAILED ^[[39m: Timeout
2024-06-13 11:58:41,837 - twister - ERROR - see: ^[[33m/home/suram/nxp/mimxrt1180_evk_cm33/temp/mimxrt1180_evk_mimxrt1189_cm33_twister_test_only/twister-out/mimxrt1180_evk_mimxrt1189_cm33/tests/posix/env/portability.posix.env.newlib/device.log^[[39m
2024-06-13 11:58:41,839 - twister - DEBUG - run test: mimxrt1180_evk/mimxrt1189/cm33/tests/posix/env/portability.posix.env.minimal
2024-06-13 11:58:41,839 - twister - DEBUG - Reset instance status from 'None' to None before run.
2024-06-13 11:58:41,839 - twister - DEBUG - Using serial device /dev/serial/by-id/usb-NXP_Semiconductors_MCU-LINK_on-board__r1E2__CMSIS-DAP_V3.122_ZLI0KV4QTLX3B-if02 @ 115200 baud
2024-06-13 11:58:41,840 - twister - DEBUG - Flash command: ['west', 'flash', '--skip-rebuild', '-d', '/home/suram/nxp/mimxrt1180_evk_cm33/temp/mimxrt1180_evk_mimxrt1189_cm33_twister_test_only/twister-out/mimxrt1180_evk_mimxrt1189_cm33/tests/posix/env/portability.posix.env.minimal', '--runner', 'linkserver', '--probe=ZLI0KV4QTLX3B']
2024-06-13 11:58:52,421 - twister - DEBUG - -- west flash: using runner linkserver
-- runners.linkserver: LinkServer: /usr/local/LinkServer/LinkServer, version v1.5.30

2024-06-13 11:58:52,422 - twister - DEBUG - halted
2024-06-13 11:58:52,424 - twister - DEBUG - Expected suite names:['env']
2024-06-13 11:58:52,424 - twister - DEBUG - Detected suite names:[]
2024-06-13 11:58:52,424 - twister - DEBUG - run status: mimxrt1180_evk/mimxrt1189/cm33/tests/posix/env/portability.posix.env.minimal error
2024-06-13 11:58:52,426 - twister - ERROR - mimxrt1180_evk/mimxrt1189/cm33 tests/posix/env/portability.posix.env.minimal      ^[[31m ERROR ^[[39m: Device issue (Flash error?)
2024-06-13 11:58:52,426 - twister - ERROR - see: ^[[33m/home/suram/nxp/mimxrt1180_evk_cm33/temp/mimxrt1180_evk_mimxrt1189_cm33_twister_test_only/twister-out/mimxrt1180_evk_mimxrt1189_cm33/tests/posix/env/portability.posix.env.minimal/device.log^[[39m
2024-06-13 11:58:52,426 - twister - DEBUG - run test: mimxrt1180_evk/mimxrt1189/cm33/tests/posix/env/portability.posix.env
```

lucien-nxp commented:
> besides, build failure. there are many init failures.
> 
> and below application will cause the rt1180 locked
> 
> tests/posix/env/portability.posix.env.newlib
> 
> ```
> 
> 2024-06-13 11:58:41,837 - twister - ERROR - mimxrt1180_evk/mimxrt1189/cm33 tests/posix/env/portability.posix.env.newlib       ^[[31m FAILED ^[[39m: Timeout
> 2024-06-13 11:58:41,837 - twister - ERROR - see: ^[[33m/home/suram/nxp/mimxrt1180_evk_cm33/temp/mimxrt1180_evk_mimxrt1189_cm33_twister_test_only/twister-out/mimxrt1180_evk_mimxrt1189_cm33/tests/posix/env/portability.posix.env.newlib/device.log^[[39m
> 2024-06-13 11:58:41,839 - twister - DEBUG - run test: mimxrt1180_evk/mimxrt1189/cm33/tests/posix/env/portability.posix.env.minimal
> 2024-06-13 11:58:41,839 - twister - DEBUG - Reset instance status from 'None' to None before run.
> 2024-06-13 11:58:41,839 - twister - DEBUG - Using serial device /dev/serial/by-id/usb-NXP_Semiconductors_MCU-LINK_on-board__r1E2__CMSIS-DAP_V3.122_ZLI0KV4QTLX3B-if02 @ 115200 baud
> 2024-06-13 11:58:41,840 - twister - DEBUG - Flash command: ['west', 'flash', '--skip-rebuild', '-d', '/home/suram/nxp/mimxrt1180_evk_cm33/temp/mimxrt1180_evk_mimxrt1189_cm33_twister_test_only/twister-out/mimxrt1180_evk_mimxrt1189_cm33/tests/posix/env/portability.posix.env.minimal', '--runner', 'linkserver', '--probe=ZLI0KV4QTLX3B']
> 2024-06-13 11:58:52,421 - twister - DEBUG - -- west flash: using runner linkserver
> -- runners.linkserver: LinkServer: /usr/local/LinkServer/LinkServer, version v1.5.30
> 
> 2024-06-13 11:58:52,422 - twister - DEBUG - halted
> 2024-06-13 11:58:52,424 - twister - DEBUG - Expected suite names:['env']
> 2024-06-13 11:58:52,424 - twister - DEBUG - Detected suite names:[]
> 2024-06-13 11:58:52,424 - twister - DEBUG - run status: mimxrt1180_evk/mimxrt1189/cm33/tests/posix/env/portability.posix.env.minimal error
> 2024-06-13 11:58:52,426 - twister - ERROR - mimxrt1180_evk/mimxrt1189/cm33 tests/posix/env/portability.posix.env.minimal      ^[[31m ERROR ^[[39m: Device issue (Flash error?)
> 2024-06-13 11:58:52,426 - twister - ERROR - see: ^[[33m/home/suram/nxp/mimxrt1180_evk_cm33/temp/mimxrt1180_evk_mimxrt1189_cm33_twister_test_only/twister-out/mimxrt1180_evk_mimxrt1189_cm33/tests/posix/env/portability.posix.env.minimal/device.log^[[39m
> 2024-06-13 11:58:52,426 - twister - DEBUG - run test: mimxrt1180_evk/mimxrt1189/cm33/tests/posix/env/portability.posix.env
> ```

Running issue will be updated later.

decsny commented:
@lucien-nxp please fix merge conflict

hakehuang commented on boards/nxp/mimxrt1180_evk/board.cmake:
```diff
@@ -0,0 +1,23 @@
+# Details about the usage of SPT on MIMXRT1180-EVK board can be referred on chapter 7 of getting start with Mcuxpresso SDK for MIMXRT1180-EVK doc in SDK package.
+if(CONFIG_SOC_MIMXRT1189_CM33)
+board_runner_args(linkserver  "--device=MIMXRT1189xxxxx:MIMXRT1180-EVK")
+board_runner_args(jlink "--device=MIMXRT1189xxx8_M33" "--reset-after-load" "--jlinkscriptfile=${RT1180_BOARD_DIR}/jlinkscript/evkmimxrt1180_cm33.jlinkscript")
```
cat /home/suram/nxp/mimxrt1180_evk_cm33/temp/mimxrt1180_evk_mimxrt1189_cm33_twister_test_only/twister-out/mimxrt1180_evk_mimxrt1189_cm33/samples/application_development/external_lib/sample.app_dev.external_lib/device.log
FATAL ERROR: runner jlink received unknown arguments: ['--jlinkscriptfile=/home/jenkins/agent/workspace/zephyr_test_pipe/modules/hal/nxp/mcux/mcux-sdk/boards/evkmimxrt1180/jlinkscript/evkmimxrt1180_cm33.jlinkscript']

lucien-nxp replied:
danieldegrasse suggest me to split jlink.py to another PR. So the issue is caused by here.
New PR created：https://github.com/zephyrproject-rtos/zephyr/pull/74226

danieldegrasse replied:
It appears there are outstanding comments on that PR. Is is possible to use `--tool-opt` as was suggested there?

lucien-nxp replied:
Sure, I am working for resolving the timeout issue. Once resolved, I will try to use `--tool-opt` , if ok, I will push together.

lucien-nxp replied:
I have modified to use '--tool-opt' to add script for cm7 core.

lucien-nxp commented:
> besides, build failure. there are many init failures.
> 
> and below application will cause the rt1180 locked
> 
> tests/posix/env/portability.posix.env.newlib
> 
> ```
> 
> 2024-06-13 11:58:41,837 - twister - ERROR - mimxrt1180_evk/mimxrt1189/cm33 tests/posix/env/portability.posix.env.newlib       ^[[31m FAILED ^[[39m: Timeout
> 2024-06-13 11:58:41,837 - twister - ERROR - see: ^[[33m/home/suram/nxp/mimxrt1180_evk_cm33/temp/mimxrt1180_evk_mimxrt1189_cm33_twister_test_only/twister-out/mimxrt1180_evk_mimxrt1189_cm33/tests/posix/env/portability.posix.env.newlib/device.log^[[39m
> 2024-06-13 11:58:41,839 - twister - DEBUG - run test: mimxrt1180_evk/mimxrt1189/cm33/tests/posix/env/portability.posix.env.minimal
> 2024-06-13 11:58:41,839 - twister - DEBUG - Reset instance status from 'None' to None before run.
> 2024-06-13 11:58:41,839 - twister - DEBUG - Using serial device /dev/serial/by-id/usb-NXP_Semiconductors_MCU-LINK_on-board__r1E2__CMSIS-DAP_V3.122_ZLI0KV4QTLX3B-if02 @ 115200 baud
> 2024-06-13 11:58:41,840 - twister - DEBUG - Flash command: ['west', 'flash', '--skip-rebuild', '-d', '/home/suram/nxp/mimxrt1180_evk_cm33/temp/mimxrt1180_evk_mimxrt1189_cm33_twister_test_only/twister-out/mimxrt1180_evk_mimxrt1189_cm33/tests/posix/env/portability.posix.env.minimal', '--runner', 'linkserver', '--probe=ZLI0KV4QTLX3B']
> 2024-06-13 11:58:52,421 - twister - DEBUG - -- west flash: using runner linkserver
> -- runners.linkserver: LinkServer: /usr/local/LinkServer/LinkServer, version v1.5.30
> 
> 2024-06-13 11:58:52,422 - twister - DEBUG - halted
> 2024-06-13 11:58:52,424 - twister - DEBUG - Expected suite names:['env']
> 2024-06-13 11:58:52,424 - twister - DEBUG - Detected suite names:[]
> 2024-06-13 11:58:52,424 - twister - DEBUG - run status: mimxrt1180_evk/mimxrt1189/cm33/tests/posix/env/portability.posix.env.minimal error
> 2024-06-13 11:58:52,426 - twister - ERROR - mimxrt1180_evk/mimxrt1189/cm33 tests/posix/env/portability.posix.env.minimal      ^[[31m ERROR ^[[39m: Device issue (Flash error?)
> 2024-06-13 11:58:52,426 - twister - ERROR - see: ^[[33m/home/suram/nxp/mimxrt1180_evk_cm33/temp/mimxrt1180_evk_mimxrt1189_cm33_twister_test_only/twister-out/mimxrt1180_evk_mimxrt1189_cm33/tests/posix/env/portability.posix.env.minimal/device.log^[[39m
> 2024-06-13 11:58:52,426 - twister - DEBUG - run test: mimxrt1180_evk/mimxrt1189/cm33/tests/posix/env/portability.posix.env
> ```
Could you help re-test the timeout cases?  I have resolved the issue, however, I only tested some of them and tested passed on my local.

hakehuang commented:
all tests/kernel/mem_protect cases ffailure

```
*** Booting Zephyr OS build v3.7.0-rc1-473-g3b9a5cc0c953 ***
Running TESTSUITE syscalls
===================================================================
START - test_arg64
E: ***** BUS FAULT *****
E:   Instruction bus error
E: r0/a1:  0x30000360  r1/a2:  0x00000001  r2/a3:  0x38017fd0
E: r3/a4:  0x00000000 r12/ip:  0x00000003 r14/lr:  0x3800e8d1
E:  xpsr:  0x21000000
E: Faulting instruction address (r15/pc): 0x3800d4c2
E: >>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0
E: Current thread: 0x30000360 (test_arg64)
Caught system error -- reason 0
Unexpected fault during test
===================================================================
RunID: 47dfb941f63ac76b47187139210a1672
PROJECT EXECUTION FAILED
```

lucien-nxp commented:
> all tests/kernel/mem_protect cases ffailure
> 
> ```
> *** Booting Zephyr OS build v3.7.0-rc1-473-g3b9a5cc0c953 ***
> Running TESTSUITE syscalls
> ===================================================================
> START - test_arg64
> E: ***** BUS FAULT *****
> E:   Instruction bus error
> E: r0/a1:  0x30000360  r1/a2:  0x00000001  r2/a3:  0x38017fd0
> E: r3/a4:  0x00000000 r12/ip:  0x00000003 r14/lr:  0x3800e8d1
> E:  xpsr:  0x21000000
> E: Faulting instruction address (r15/pc): 0x3800d4c2
> E: >>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0
> E: Current thread: 0x30000360 (test_arg64)
> Caught system error -- reason 0
> Unexpected fault during test
> ===================================================================
> RunID: 47dfb941f63ac76b47187139210a1672
> PROJECT EXECUTION FAILED
> ```

Hi hake, 
mem_protect cases issue have been resolved, could you help re-test again. It caused by permission controller trng reject the cm33 access flexspi region.

hakehuang commented:
> Hi hake,
> mem_protect cases issue have been resolved, could you help re-test again. It caused by permission controller trng reject the cm33 access flexspi region.

running test, will feedback once ready.

lucien-nxp commented:
> > Hi hake,
> > mem_protect cases issue have been resolved, could you help re-test again. It caused by permission controller trng reject the cm33 access flexspi region.
> 
> running test, will feedback once ready.
> 
> ```
> tests/arch/arm/arm_irq_vector_table/arch.arm.irq_vector_table
> 
> tests/drivers/can/shell/drivers.can.shell
> 
> tests/drivers/uart/uart_async_rx/drivers.uart.async_rx
> 
> tests/kernel/interrupt/arch.shared_interrupt
> 
> tests/kernel/mutex/mutex_api/kernel.mutex
> 
> tests/kernel/sleep/kernel.common.timing
> 
> tests/kernel/usage/thread_runtime_stats/kernel.usage
> 
> tests/kernel/timer/timer_behavior/kernel.timer.timer
> 
> tests/kernel/timer/timer_monotonic/kernel.timer.monotonic
> 
> tests/kernel/timer/timer_api/kernel.timer
> 
> tests/kernel/workq/work/kernel.workqueue.api
> 
> tests/kernel/sched/schedule_api/kernel.scheduler
> 
> tests/kernel/sched/schedule_api/kernel.scheduler.slice_perthread
> ```
> 
> all above case failed

I have resolved the issue and test these 12 cases passed on my local. Could you help start a new test cycle? Thank you,

hakehuang commented on drivers/gpio/gpio_mcux_rgpio.c:
```diff
@@ -80,6 +80,51 @@ static int mcux_rgpio_configure(const struct device *dev,
+			}
+		} else {
+			/* Set pin to no pull */
+			reg |= IOMUXC_SW_PAD_CTL_PAD_PUS_MASK;
```
this is wrong, see https://github.com/zephyrproject-rtos/zephyr/pull/75328

lucien-nxp replied:
Fixed, thank you for reminding.

hakehuang commented:
@lucien-nxp , now only this case need check
```
tests/arch/arm/arm_irq_vector_table/arch.arm.irq_vector_table
```

hakehuang commented:
No CI for this board is clean.

decsny commented on soc/nxp/imxrt/Kconfig:
```diff
@@ -96,6 +97,28 @@ config FLEXSPI_CONFIG_BLOCK_OFFSET
 	  sequence (optional), etc. The boot ROM expects FlexSPI configuration
 	  parameter to be presented in serial nor flash.
 
+config IMAGE_CONTAINER_OFFSET
```
is this different from IVT on the other RT parts?

lucien-nxp replied:
Yes， RT1180 is different with other RT platforms.

dleach02 replied:
@lucien-nxp do we document the details someplace on this?

decsny commented on soc/nxp/imxrt/imxrt118x/Kconfig:
```diff
@@ -0,0 +1,63 @@
+	select CPU_HAS_ARM_SAU if SOC_MIMXRT1189_CM33
+	select HAS_MCUX
+	select HAS_MCUX_CCM_REV2
+	select HAS_MCUX_LPUART
```
dont think this is needed

lucien-nxp replied:
Yes, I will delete HAS_MCUX_LPUART

decsny commented on boards/nxp/mimxrt1180_evk/mimxrt1180_evk_mimxrt1189_cm7.yaml:
```diff
@@ -0,0 +1,20 @@
+  - gnuarmemb
+  - xtools
+ram: 256
+flash: 256
```
only 256? is that right?

lucien-nxp replied:
Yes, it's right. Because the cm7 core image is a ram target image like RT1170_CM4, I checked the flash size described in cm4 yml is 128.

decsny reviewed (changes requested):
please squash the manifest and linker commit to the soc commit

lucien-nxp commented on soc/nxp/imxrt/imxrt118x/Kconfig:
```diff
@@ -0,0 +1,63 @@
+config BYPASS_LDO_LPSR
+	bool "Bypass LDO lpsr"
+
+config ADJUST_LDO
```
Because the architecture of the entire RT118X is different from that of RT11XX, it is not compatible. For example, RT11XX is a dual-core chip combined with CM7 and CM4, but RT118X is a dual-core chip combined with CM33 and CM7, and CM33 can only be the master core

lucien-nxp replied:
> these configs are copied from the RT11XX. We should reuse code instead of copying. Also, why can't RT118x be in the RT11xx folder for more code and config reuse.

These kconfig don't be used on RT1180 soc.c file, I will delete them.

lucien-nxp commented on boards/nxp/mimxrt1180_evk/Kconfig.defconfig:
```diff
@@ -0,0 +1,12 @@
+
+if BOARD_MIMXRT1180_EVK
+
+config SYS_CLOCK_HW_CYCLES_PER_SEC
```
Yes, the content is duplicated, I will delete it.

lucien-nxp commented on drivers/gpio/gpio_mcux_rgpio.c:
```diff
@@ -80,6 +80,51 @@ static int mcux_rgpio_configure(const struct device *dev,
+			reg |= IOMUXC_SW_PAD_CTL_PAD_PUE_MASK;
+			if (((flags & GPIO_PULL_UP) != 0)) {
+				reg |= IOMUXC_SW_PAD_CTL_PAD_PUS_MASK;
+			} else {
```
rgpio driver have been updated according to your suggestion, thank you.

lucien-nxp commented on drivers/pinctrl/pinctrl_imx.c:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2022 NXP
+ * Copyright (c) 2022-2024 NXP
```
Fixed.

lucien-nxp commented on boards/nxp/mimxrt1180_evk/board.cmake:
```diff
@@ -0,0 +1,23 @@
+#
+# Copyright 2024, NXP
```
Fixed

lucien-nxp commented on boards/nxp/mimxrt1180_evk/doc/index.rst:
```diff
@@ -0,0 +1,251 @@
+
+.. code-block:: console
+
+   ***** Booting Zephyr OS v3.6.0-xxx-xxxxxxxxxxxxx *****
```
Fixed

lucien-nxp commented:
> please squash the manifest and linker commit to the soc commit

Squashed.

hakehuang commented:
the CI failure is related to https://github.com/zephyrproject-rtos/zephyr/issues/76456 not this PR issue.

lucien-nxp commented:
> the CI failure is related to #76456 not this PR issue.

https://github.com/zephyrproject-rtos/zephyr/pull/76461 addressed the issue, I have rebased my PR to newest commit.

lucien-nxp commented:
David inform me he has prepared sdk2.16.0 update on hal_nxp repo, change version recorded into west.yml to new PR. hal_nxp/pull/415.

dleach02 commented on west.yml:
```diff
@@ -198,7 +198,7 @@ manifest:
         - hal
     - name: hal_nxp
-      revision: 272f84ca60afc759d81ed94100e78b37a859d3ff
+      revision: pull/391/head
```
Update this to point to HAL PR 415

lucien-nxp replied:
Yes, I know. I have modified to 415, however, some cases build failed with 415PR. https://github.com/zephyrproject-rtos/zephyr/actions/runs/10195617084/job/28206275931
I want to get a successful result based on 391 firstly, then I will change revision to 415 tomorrow.

lucien-nxp replied:
Updated to PR 415.

dleach02 commented on soc/nxp/imxrt/CMakeLists.txt:
```diff
@@ -37,6 +37,24 @@ if(CONFIG_SOC_SERIES_IMXRT10XX OR CONFIG_SOC_SERIES_IMXRT11XX)
   zephyr_compile_definitions(XIP_EXTERNAL_FLASH)
 endif()
 
+if(CONFIG_SOC_SERIES_IMXRT118X)
```
Where should we document this stuff? Is there a link to some NXP documents on the various sections?

dleach02 commented:
@lucien-nxp I updated your PR to remove the west update now that SDK2.16.0 has merged
